pkg/config/process.go

```diff
@@ -70,3 +94,48 @@ func setupProcesses(config Config) {
 	)
 	procBindEnvAndSetDefault(config, "process_config.process_discovery.interval", 4*time.Hour)
 }
+
+var displayProcConfigEnabledWarningOnce sync.Once
+
+// displayProcConfigEnabledDeprecationWarning displays a deprecation warning for process_config.enabled only once.
+// For testing purposes, it rethrows the warning.
+func displayProcConfigEnabledDeprecationWarning() (warning error) {
+	displayProcConfigEnabledWarningOnce.Do(func() {
+		warning = log.Warn("process_config.enabled is deprecated, use process_config.container_collection.enabled" +
+			" and process_config.process_collection.enabled instead")
+	})
+	return
+}
+
+// GetContainerCollectionEnabled retrieves the value of process_config.container_collection.enabled.
+// If process_config.enabled is set, we display a deprecation warning and use that value instead.
+func GetContainerCollectionEnabled(config Config) bool {
+	if config.IsSet("process_config.enabled") {
+		_ = displayProcConfigEnabledDeprecationWarning()
+
+		procConfigEnabled := strings.ToLower(config.GetString("process_config.enabled"))
+		if procConfigEnabled == "disabled" {
+			return false
+		}
+		result, _ := strconv.ParseBool(procConfigEnabled)
```

Comment:
nit: what do you think of using `enabled` instead of `result` and add a comment like the one from `pkg/process/config/yaml_config.go` to make it more explicit how we're mapping `process_config.enabled` to the new settings?
```
		// A string indicate the enabled state of the Agent.
		//   If "false" (the default) we will only collect containers.
		//   If "true" we will collect containers and processes.
		//   If "disabled" the agent will be disabled altogether and won't start.
```